integration/query_comparison_compat_test.go

```diff
@@ -86,6 +86,32 @@ func testQueryComparisonCompatImplicit() map[string]queryCompatTestCase {
 			filter:         bson.D{{"v", math.SmallestNonzeroFloat64}},
 			resultPushdown: true,
 		},
+
+		"DoubleBigInt64": {
+			filter:         bson.D{{"v", float64(2 << 61)}},
+			resultPushdown: true,
+			skip:           "https://github.com/FerretDB/FerretDB/issues/2057",
+		},
+		"DoubleBigInt64PlusOne": {
+			filter:         bson.D{{"v", float64(2<<61 + 1)}},
+			resultPushdown: true,
+			skip:           "https://github.com/FerretDB/FerretDB/issues/2057",
+		},
+		"Int64Max": {
+			filter:         bson.D{{"v", int64(math.MaxInt64)}},
+			resultPushdown: true,
+		},
+		"Int64Min": {
+			filter:         bson.D{{"v", int64(math.MinInt64)}},
+			resultPushdown: true,
+			skip:           "https://github.com/FerretDB/FerretDB/issues/2057",
+		},
+		"Int64DoubleBig": {
+			filter:         bson.D{{"v", int64(2 << 60)}},
+			resultPushdown: true,
+			skip:           "https://github.com/FerretDB/FerretDB/issues/2057",
+		},
```

Comment:
Do you have a test for string being regular expression? The Implicit and $eq behave differently on that.

Reply:
Resolved on pair programming!

Reply:
For anyone interested, implicit and $eq is different when the filter value is regular expression. Since we do not support regular expression pushdown yet, nothing need to be address on this. I've learnt this too